### PR TITLE
e2e: Add Agent command arguments

### PR DESCRIPTION
**Pull request by Kaderinho**
### What does this PR do?

Adds a way to give arguments to Agent commands. It also removes `require.NoError` which was making tests panicking when the Agent was not ready / returns error code.

### Motivation

Be able to test more cases

### Additional Notes

- Since e2e is for internal / devs usage I did not do any argument sanitizing (since `Execute` is available, I don't think it matters)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/utils/e2e/client/agent.go [hunk 1]
```diff
@@ -41,16 +41,24 @@ func (agent *Agent) initService(t *testing.T, data *agent.ClientData) error {
...
 func (agent *Agent) GetCommand(parameters string) string {
 	return agent.os.GetRunAgentCmd(parameters)
 }
 
-func (agent *Agent) Config() string {
-	return agent.vmClient.Execute(agent.GetCommand("config"))
+func (agent *Agent) executeCommand(command string, commandArgs ...AgentArgsOption) string {
+	args, err := newAgentArgs(commandArgs...)
+	if err != nil {
+		return ""
```
Comment (pducolin) [inline test/new-e2e/utils/e2e/client/agent.go:51]: **💭 thought**
Not a big fan of hiding errors, I'd rather use the `ExecuteWithError` for `WaitForReady` reusing `newStatus`, something like

```
func (agent *Agent) WaitForReady() err {
   err := loopOnError({
     statusOutput, err := agent.ExecuteWithError("status")
     if err {
       return err
     }
     status := newStatus(statusOutput)
     if !status.isReady() {
       return new error
     }
     return nil
  })
  return err
}

Comment (Kaderinho) [reply test/new-e2e/utils/e2e/client/agent.go:51]: I'm not a fan either, and I realised I forgot to talk about it in the PR description.

My thoughts was that we should return (string, error) for all the Agent commands (I already have silent error; e.g. got an empty string from a failing command) and use `ExecuteWithError` within `executeCommand()`.

For the WaitForReady() example you gave, do you want me to apply it on this PR and re introduce `requireNoError` in `Execute` ?

Comment (pducolin) [reply test/new-e2e/utils/e2e/client/agent.go:51]: Yes please !

[the remark this answers is not in the document]
Comment (Kaderinho) [reply test/new-e2e/utils/e2e/client/agent.go]: I removed the error in the return type of newAgentArgs: fb0048b10f40ff9a9b6ce358bf26ae059a781127

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: test/new-e2e/utils/e2e/client/agent.go [hunk 2]
```diff
@@ -90,13 +95,19 @@ func (a *Agent) WaitForReadyTimeout(timeout time.Duration) error {
...
+		}
+
+		isReady, err := newStatus(statusOutput).isReady()
 		if err != nil {
 			return err
 		}
 		if !isReady {
 			return errors.New("agent not ready")
 		}
+
```
Comment (pducolin) [inline test/new-e2e/utils/e2e/client/agent.go:110]: **👏 praise**
Thank you !

**Review by pducolin — APPROVED** [review]
Thank you so much for the fix and for addressing comments !

**Pull request merged**